src/components/Captcha.vue

```diff
@@ -9,68 +9,109 @@
-                if (this.recaptchaResponse) {
-                    event.network.ircClient.raw('CAPTCHA', this.recaptchaResponse);
-                }
-            });
-        });
+        this.siteId = options.captchaSiteId || options.recaptchaSiteId || '';
+        this.url = options.captchaUrl || options.recaptchaUrl;
+        this.url = this.url || 'https://www.google.com/recaptcha/api.js';
 
         this.listen(this.$state, 'irc.raw.CAPTCHA', (command, event, network) => {
-            if (network !== this.network) {
+            if (event.params[0] !== 'NEEDED') {
                 return;
             }
 
-            if (event.params[0] === 'NEEDED') {
-                this.loadRecaptcha();
+            if (this.response) {
+                // We already have a valid response
+                this.sendResponse(network);
+                return;
             }
+
+            this.addNetwork(network);
+            this.showCaptcha();
         });
     },
     methods: {
-        loadRecaptcha() {
-            this.showCaptcha = true;
+        showCaptcha() {
+            this.visible = true;
+
+            if (this.loaded) {
+                return;
+            }
+            this.loaded = true;
 
-            // Recaptcha calls this callback once it's loaded and ready to be used
-            window.recaptchaLoaded = () => {
-                window.grecaptcha.render(this.$refs.captchacontainer, {
-                    sitekey: this.recaptchaSiteId,
-                    callback: this.recaptchaSuccess,
-                    'expired-callback': this.recaptchaExpired,
-                });
-            };
+            this.$nextTick(() => {
+                // Recaptcha calls this callback once it's loaded and ready to be used
+                window.recaptchaLoaded = () => {
+                    window.grecaptcha.render(this.$refs.container, {
+                        sitekey: this.siteId,
+                        callback: this.captchaSuccess,
+                        'expired-callback': this.captchaExpired,
+                    });
+                };
 
-            let scr = document.createElement('script');
-            scr.src = this.recaptchaUrl + '?onload=recaptchaLoaded&render=explicit';
-            scr.defer = true;
-            this.$el.appendChild(scr);
+                let scr = document.createElement('script');
+                scr.src = this.url + '?onload=recaptchaLoaded&render=explicit';
+                scr.defer = true;
+                this.$el.appendChild(scr);
+            });
         },
-        recaptchaSuccess(response) {
-            this.recaptchaResponse = response;
+        captchaSuccess(response) {
+            this.response = response;
 
-            // If we have a network instance already, send the captcha response
-            if (this.network && this.network.state === 'connecting') {
-                this.network.ircClient.raw('CAPTCHA', response);
+            this.networks.forEach((network) => {
```

Comment:
This part seems a little weird. Each network captcha request would have a difference response wouldn't they? So sending the same response to every network would cause them to fail?

Reply:
WIG only uses one captcha key, so it should be fine for multiple networks to validate the same response,

i shall try to find a way to test it

Reply:
then again i might be totally wrong :(

need to look into this some more